components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/token/AccessTokenIssuer.java

```diff
@@ -551,25 +552,37 @@ private boolean validateScope(OAuthTokenReqMessageContext tokReqMsgCtx) throws I
             }
             return isValidScope;
         }
-        List<String> allowedScopes = OAuthServerConfiguration.getInstance().getAllowedScopes();
+        boolean isManagementApp = getServiceProvider(tokenReqDTO).isManagementApp();
         List<String> requestedAllowedScopes = new ArrayList<>();
+        String[] authorizedInternalScopes = new String[0];
         String[] requestedScopes = tokReqMsgCtx.getScope();
-        List<String> scopesToBeValidated = new ArrayList<>();
-
-        if (requestedScopes != null) {
-            for (String scope : requestedScopes) {
-                if (OAuth2Util.isAllowedScope(allowedScopes, scope)) {
-                    requestedAllowedScopes.add(scope);
-                } else {
-                    scopesToBeValidated.add(scope);
+        if (GrantType.CLIENT_CREDENTIALS.toString().equals(grantType) && !isManagementApp) {
+            log.debug("Application is not configured as Management App and the grant type is client credentials. " +
+                    "Hence skipping internal scope validation to stop issuing internal scopes.");
+        } else {
```

Comment:
```suggestion
        if (GrantType.CLIENT_CREDENTIALS.toString().equals(grantType)){
        
         if  (!isManagementApp) {
         // Add debug logs
         } else {
             if (ArrayUtils.contains(requestedScopes, INTERNAL_LOGIN_SCOPE)){
               // Logic
             }
             // Logic
             if (ArrayUtils.contains(requestedScopes, SYSTEM_SCOPE)){
               // Logic
             }
         }
         
         }
```
Shall we simplify the if condition?

Reply:
Can't simplify this because when removing the internal login scope from the requested scope we don't have the authorized internal scope detail. To calculate the authorized internal scope we need the requested scopes without internal_login.